### PR TITLE
[FEAT] Added l10n_fi_payment_reference module.

### PR DESCRIPTION
**Pull request by n17**

### CODE REVIEW

**Comment by n17** [comment]
I have signed the CLA but it seems to be pending still.

File: l10n_fi_payment_reference/README.rst [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+The module allows users to generate a valid invoice payment reference for sales invoices.
+
+.. contents::
+   :local:
+
+Configuration
+=============
+
+Payment Reference Type can be selected from Account Settings.
```
Comment (mlaitinen) [inline l10n_fi_payment_reference/README.rst:33]: @n17 The `views/res_config.xml` seems to be missing which means that this setting is not really visible in the Accounting settings. Could you please check it? https://github.com/avoinsystems/l10n_fi/blob/11.0/views/res_config.xml

Comment (mlaitinen) [reply l10n_fi_payment_reference/README.rst:33]: Same for `views/account_invoice.xml`: https://github.com/avoinsystems/l10n_fi/blob/11.0/views/account_invoice.xml

I can't see the payment reference in the invoice view.

**Review by mlaitinen — CHANGES_REQUESTED** [review]
Need the views

File: l10n_fi_payment_reference/README.rst (hunk 1 above)
Comment (n17) [reply l10n_fi_payment_reference/README.rst:33]: @mlaitinen Missing xml files are now added.

**Review by mlaitinen — APPROVED** [review]

**Comment by mlaitinen** [comment]
@aisopuro or/and @jarmokortetjarvi, feel like reviewing?

File: l10n_fi_payment_reference/models/account_invoice.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        for invoice in self:
+            ref_type = invoice.payment_reference_type
+            if invoice.payment_reference or not ref_type or ref_type == 'none':
+                continue
+
+            if invoice.number:
+                method_name = '_compute_payment_reference_' + ref_type
+                if not hasattr(invoice, method_name):
+                    raise NotImplementedError(
+                        "Payment reference type {} doesn't have"
```
Comment (jarmokortetjarvi) [inline l10n_fi_payment_reference/models/account_invoice.py]: Isn't this untranslatable? This very minor issue, as this error is unlikely to occur

File: l10n_fi_payment_reference/views/account_invoice.xml [hunk 3]
```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+        <!-- Sales invoice & refund form -->
+        <record id="invoice_form_finnish" model="ir.ui.view">
+            <field name="name">account.invoice.form.finnish</field>
+            <field name="model">account.invoice</field>
+            <field name="inherit_id" ref="account.invoice_form"/>
+            <field name="arch" type="xml">
+                <data>
```
Comment (jarmokortetjarvi) [inline l10n_fi_payment_reference/views/account_invoice.xml]: Is this `<data>` necessary?

File: l10n_fi_payment_reference/views/account_invoice.xml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+            </field>
+        </record>
+
+        <!-- Purchase invoice & refund form -->
+        <record id="invoice_supplier_form_finnish" model="ir.ui.view">
+            <field name="name">account.supplier.invoice.form.finnish</field>
+            <field name="model">account.invoice</field>
+            <field name="inherit_id" ref="account.invoice_supplier_form"/>
+            <field name="arch" type="xml">
+                <data>
```
Comment (jarmokortetjarvi) [inline l10n_fi_payment_reference/views/account_invoice.xml]: Is this `<data>` necessary?

**Review by jarmokortetjarvi — APPROVED** [review]
Looks good to me. Added a few comments outside of review, as none of them aren't really a reason to not approve. View indentation could be refactored before merge (especially if data-tags are removed)

**Comment by aisopuro** [comment]
@mlaitinen or @jarmokortetjarvi , you've both accepted this PR it seems, could one of you merge it (I don't have access)?

**Pull request merged**